src/backends/perf_pt/decode.c

```diff
@@ -169,8 +169,10 @@ perf_pt_init_block_decoder(void *buf, uint64_t len, int vdso_fd, char *vdso_file
 bool
 perf_pt_next_block(struct pt_block_decoder *decoder, int *decoder_status,
                    uint64_t *addr) {
-    // If there are events pending, drain them first.
-    *decoder_status = drain_events(decoder, *decoder_status);
+    // If there are events pending, look at those first.
+    if (handle_events(decoder, decoder_status) != true) {
```

Comment:
This seems like a slightly odd reason for returning `false`?

Reply:
Perhaps this should be an `assert`?

Reply:
All of the cases I don't expect to see are already caught by an assert in the callee. Returning false here only applies to buffer overflow or `pt_blk_event()` failing.